src/Tuura/Concept/Circuit/Basic.hs

```diff
@@ -35,17 +36,19 @@ data Concept s e a = Concept
                    {
                        initial   :: a -> InitialValue,
                        arcs      :: [([e], e)],
-                       interface :: a -> Interface
+                       interface :: a -> Interface,
+                       invariant :: [[e]]
                    }
```

Comment:
So, this is a list of lists of events. What is its meaning? Maybe it's worth adding a proper datatype for better clarity. For example, `invariant :: [Invariant e]`, that is, we have a list of `Invariant` descriptions. 

In fact, we could do something similar with `arcs` at some point. For example, `arcs :: [Causality e]`, and then define `data Causality e = AndCausality e e | OrCausality [e] e`. That would likely make the code cleaner and self-explanatory.